lib/webpacker/node_bundler.rb

```diff
@@ -0,0 +1,67 @@
+# Finds available global or local node package manager - yarn or npm
+
+require "webpacker/configuration"
+
+module Webpacker::NodeBundler
+  extend self
+
+  def name
+    @package ||= packages.find(-> { raise Errno::ENOENT }) { |package| available?(package) }
+
+  rescue Errno::ENOENT
+    $stderr.puts "Node package managers npm and yarn not found! Scanned following paths for executables - #{packages}"
+    $stderr.puts "Make sure either npm or yarn is installed globally, or locally within your app node_modules folder."
+    exit!
+  end
+
+  def command
+    commands[name]
+  end
+
+  def command_dev
+    commands[:dev][name]
+  end
+
+  private
+
+    def available?(package)
+      null_device = Gem.win_platform? ? "/nul 2>&1" : "/dev/null 2>&1"
+      system("#{package} --version > #{null_device}")
```

Comment:
"surely" might be a bit strong - I work on hundreds of projects and don't have yarn installed for any of them.

I also think the presence of yarn - which perhaps one package I work on might require - shouldn't force it to use yarn. Explicit configuration when it's ambiguous might be useful.